src/Symfony/Bridge/Doctrine/PropertyInfo/DoctrineExtractor.php

```diff
@@ -122,7 +122,12 @@ public function getTypes($class, $property, array $context = [])
                             /** @var ClassMetadataInfo $subMetadata */
                             $indexProperty = $subMetadata->getSingleAssociationReferencedJoinColumnName($fieldName);
                             $subMetadata = $this->entityManager ? $this->entityManager->getClassMetadata($associationMapping['targetEntity']) : $this->classMetadataFactory->getMetadataFor($associationMapping['targetEntity']);
-                            $typeOfField = $subMetadata->getTypeOfField($indexProperty);
+
+                            //Not a property, maybe a column name?
```

Comment:
Unpopular opinion: should'nt we deprecate `colum_name` here and at line 117?

Supporting both propertyName AND columnName in both mapping and reverseMapping increase the complexity for nothing. IMHO we should support a single way to do things

Reply:
Let's not wait to resolve this question before merging this PR.